architecture/core.rst

```diff
@@ -800,42 +800,52 @@ Shared Memory
 Shared Memory is a block of memory that is shared between the non-secure and the
 secure world. It is used to transfer data between both worlds.
 
-Shared Memory Allocation
+Shared Memory References
 ========================
-The shared memory is allocated by the Linux driver from a pool ``struct
-shm_pool``, the pool contains:
 
-    - The physical address of the start of the pool
-    - The size of the pool
-    - Whether or not the memory is cached
-    - List of chunk of memory allocated.
+The shared memory is allocated and managed by the non-secure world, i.e the
+Linux OP-TEE driver. Secure world only considers the individual shared buffers,
+not their pool. Each shared memory is referenced with associated attributes:
+    - Buffer start address and byte size
+    - Cacheability of the buffer
+    - List of chunks if mapped from discontiguos pages.
 
-.. note::
-    - The shared memory pool is physically contiguous.
-    - The shared memory area is **not secure** as it is used by both non-secure
-      and secure world.
+Shared memory buffer references manipulated must fit inside one of the
+shared memory areas known from the OP-TEE core. OP-TEE supports 2 kind
+of shared memory areas: a mandatory one for physically contiguous buffers
+and  optional extra memory areas for non-contiguous buffers.
```

Comment:
s/and  optional/and optional/ (double spaces) s/non-contiguous/noncontiguous/

Reply:
thanks + ok for all `noncontiguous`